src/System.Linq/samples/project.json

```diff
@@ -0,0 +1,13 @@
+{
+  "dependencies": {
+    "System.Collections": "4.0.10",
```

Comment:
> Any dependency issues with this project would also exist in the test project

I'm not sure about that. Maybe one of the methods used in the samples project (but not in the test project) was removed or renamed, or made incompatible in some way. Anyways, I think that as long as the test and sample projects lock their dependencies together, or ensure in some way that their dependencies stay in sync, then the issue probably won't surface. I think that if it doesn't happen, though, you will probably be left with some annoying breaks somewhere down the line when we try to re-lock dependencies (or run our script to update dependencies, when * versioning goes away).

I'm also not 100% clear how project references behave with nuget dependencies, either, perhaps someone else is more familiar. If I have a NuGet dependency on System.Console 4.0-beta001 and I reference a project that has a NuGet dependency on System.Console 4.0-beta003, what happens there? If I automatically grab 4.0-beta003 then i guess this issue wouldn't ever occur.